modules/basicfuncs/ip-funcs.c

```diff
@@ -40,3 +42,121 @@ tf_ipv4_to_int(LogMessage *msg, gint argc, GString *argv[], GString *result)
 }
 
 TEMPLATE_FUNCTION_SIMPLE(tf_ipv4_to_int);
+
+typedef struct _DnsResolveIpState
+{
+  TFSimpleFuncState super;
+  HostResolveOptions host_resolve_options;
+} DnsResolveIpState;
+
+static gboolean
+_process_use_fqdn(const gchar *option_name, const gchar *value, gpointer data, GError **error)
+{
+  HostResolveOptions *host_resolve_options = (HostResolveOptions *)data;
+  host_resolve_options->use_fqdn = cfg_process_yesno(value);
+  return TRUE;
+}
+
+static gboolean
+_process_use_dns(const gchar *option_name, const gchar *value, gpointer data, GError **error)
+{
+  HostResolveOptions *host_resolve_options = (HostResolveOptions *)data;
+  host_resolve_options->use_dns = cfg_process_yesno(value);
+  return TRUE;
+}
+
+static gboolean
+_process_dns_cache(const gchar *option_name, const gchar *value, gpointer data, GError **error)
+{
+  HostResolveOptions *host_resolve_options = (HostResolveOptions *)data;
+  host_resolve_options->use_dns_cache = cfg_process_yesno(value);
+  return TRUE;
+}
+
+static gboolean
+_process_normalize_hostnames(const gchar *option_name, const gchar *value, gpointer data, GError **error)
+{
+  HostResolveOptions *host_resolve_options = (HostResolveOptions *)data;
+  host_resolve_options->normalize_hostnames = cfg_process_yesno(value);
+  return TRUE;
+}
+
+static gboolean
+_parse_host_resolve_options(gint *argc, gchar **argv[], HostResolveOptions *host_resolve_options, GError **error)
+{
+  gboolean result = FALSE;
+  GOptionEntry entries[] =
+  {
+    { "use-fqdn", 'f', 0, G_OPTION_ARG_CALLBACK, _process_use_fqdn, NULL, NULL },
+    { "use-dns", 'd', 0, G_OPTION_ARG_CALLBACK, _process_use_dns, NULL, NULL },
+    { "dns-cache", 'c', 0, G_OPTION_ARG_CALLBACK, _process_dns_cache, NULL, NULL },
+    { "normalize-hostnames", 'n', 0, G_OPTION_ARG_CALLBACK, _process_normalize_hostnames, NULL, NULL },
+    { NULL }
+  };
+
+  GOptionContext *ctx = g_option_context_new(*argv[0]);
+  GOptionGroup *group = g_option_group_new("host-resolve-options", NULL, NULL, host_resolve_options, NULL);
+  g_option_group_add_entries(group, entries);
+  g_option_context_set_main_group(ctx, group);
+
+  if (!g_option_context_parse(ctx, argc, argv, error))
+    goto exit;
+
+  result = TRUE;
+exit:
+  g_option_context_free(ctx);
+  return result;
+}
+
+static gboolean
+_setup_host_resolve_options(DnsResolveIpState *state, gint *argc, gchar **argv[], GError **error)
+{
+  host_resolve_options_global_defaults(&state->host_resolve_options);
```

Comment:
Please note that in case of `use-cache` not only the cache will be used, but the hosts file as well (that's how `dns-cache` works right now: https://github.com/syslog-ng/syslog-ng/pull/1923).

In fact in global config the `use-dns(no)` and `use-cache(yes)` combination have been eliminated, as that is probably not something that the user actually desires. Do you think it would make sense to disable here that case as well ?

Reply:
Sure, the context is the same :) I will do it, thanks for the tip!